src/main/kotlin/uk/gov/justice/digital/hmpps/personintegrationapi/common/config/WebClientConfiguration.kt

```diff
@@ -0,0 +1,98 @@
+package uk.gov.justice.digital.hmpps.personintegrationapi.common.config
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.DependsOn
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.oauth2.client.AuthorizedClientServiceOAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.InMemoryOAuth2AuthorizedClientService
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientProviderBuilder
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService
+import org.springframework.security.oauth2.client.endpoint.DefaultClientCredentialsTokenResponseClient
+import org.springframework.security.oauth2.client.endpoint.OAuth2ClientCredentialsGrantRequest
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository
+import org.springframework.security.oauth2.client.web.reactive.function.client.ServletOAuth2AuthorizedClientExchangeFilterFunction
+import org.springframework.web.context.annotation.RequestScope
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.support.WebClientAdapter
+import org.springframework.web.service.invoker.HttpServiceProxyFactory
+import uk.gov.justice.digital.hmpps.personintegrationapi.common.client.PrisonApiClient
+import uk.gov.justice.digital.hmpps.personintegrationapi.config.UserEnhancedOAuth2ClientCredentialGrantRequestConverter
+import uk.gov.justice.hmpps.kotlin.auth.healthWebClient
+import java.time.Duration
+
+@Configuration
+class WebClientConfiguration(
+  @Value("\${hmpps-auth.url}") private val authBaseUri: String,
+  @Value("\${prison-api.base_url}") private val prisonApiBaseUri: String,
+  @Value("\${api.timeout:20s}") val healthTimeout: Duration,
+  @Value("\${api.timeout:90s}") val timeout: Duration,
```

Comment:
I separated out individual timeouts for more flexibility in prison-person-api, not sure if you wanted to follow suit